level/level.go

```diff
@@ -100,6 +129,36 @@ func allowed(allowed level) Option {
 	return func(l *logger) { l.allowed = allowed }
 }
 
+// Parse a string to it's corresponding level value. Valid strings are "debug",
+// "info", "warn", and "error". Strings are normalized via strings.TrimSpace and
+// strings.ToLower.
+func Parse(level string) (Value, error) {
+	switch strings.TrimSpace(strings.ToLower(level)) {
+	case debugValue.name:
+		return debugValue, nil
+
+	case infoValue.name:
+		return infoValue, nil
+
+	case warnValue.name:
+		return warnValue, nil
+
+	case errorValue.name:
+		return errorValue, nil
+
+	default:
+		return nil, ErrInvalidLevelString
+	}
+}
+
+// ParseDefault calls Parse and returns the default Value on error.
+func ParseDefault(level string, def Value) Value {
+	if v, err := Parse(level); err == nil {
```

Comment:
Please invert this logic so that it is an `err != nil` comparison. That is by far the most common form and the one Go developer brains are conditioned to expect. Using the expected form is less likely to confuse us if we don't read it carefully enough.

Reply:
Maybe I'm wrong but this code was suggested by @peterbourgon

Reply:
Interesting. I was thinking of this:

```
v, err := Parse(level)
if err != nil {
	return def
}
return v
```
What do you think @peterbourgon?

Reply:
> Interesting. I was thinking of this:
> 
> ```
> v, err := Parse(level)
> if err != nil {
> 	return def
> }
> return v
> ```
> 
> What do you think @peterbourgon?

This looks more idiomatic Go.

Reply:
> Interesting. I was thinking of this:
> 
> ```
> v, err := Parse(level)
> if err != nil {
> 	return def
> }
> return v
> ```
> 
> What do you think @peterbourgon?

looks good to me as well, let's wait for @peterbourgon for a final decision

Reply:
No difference to me. Go for it.

Reply:
pushed the agreed change